pkg/services/store/types.go

```diff
@@ -95,3 +95,40 @@ type RootStorageMeta struct {
 
 	Config RootStorageConfig `json:"config"`
 }
+
+type StorageListFrame struct {
+	*data.Frame
+}
+
+type listFrameField string
+
+const (
+	titleListFrameField       listFrameField = "title"
+	nameListFrameField        listFrameField = "name"
+	descriptionListFrameField listFrameField = "description"
+	mediaTypeListFrameField   listFrameField = "mediaType"
+	storageTypeListFrameField listFrameField = "storageType"
+	readOnlyListFrameField    listFrameField = "readOnly"
+	builtInListFrameField     listFrameField = "builtIn"
+	sizeListFrameField        listFrameField = "size"
+)
+
+func (s *StorageListFrame) GetFileNames() []string {
```

Comment:
I tried to come up with something abstract, but, due to the lack of generics & mapped types, all other approaches resulted in more hassle for the client (casting) - not sure if there is any better alternative